src/include/Context.h

```diff
@@ -58,6 +58,7 @@ class Context {
   virtual void finish(int r) = 0;
 
  public:
+  static const int HAVE_PGLOCK = 256;
```

Comment:
nit: let's call this FLAG_SYNC indicating we are still in the submitting context.  At this layer we don't know what a pglock is or what it means.